src/web/routes/application/common/state-machine/state-machine.js

```diff
@@ -17,9 +17,33 @@ const actions = {
 
 const getStepForPath = (path, steps) => steps.find(step => step.path === path)
 
+/**
+ * Next path is navigable if the next step does not exist, if it doesn't define an isNavigable function, or that function returns true.
+ * E.g. if the current step is the end of the 'in-progress' journey, the next path will be /check. There is no step matching /check, so we assume it is navigable.
+ */
+function isNextPathNavigable (nextStep, req) {
+  return !nextStep || !nextStep.isNavigable || nextStep.isNavigable(req.session)
+}
+
+/**
+ * Ask the current step for the next path. Test whether the step matching that path is navigable. If not, ask that step for the next path; repeat.
+ * @param path the path of the current step
+ * @param req the current request
+ * @param steps the steps of the apply journey
+ * @returns the path of the next step
+ */
+const getNextPath = (path, req, steps) => {
```

Comment:
I think maybe this should be `getNextNavigablePath`, which would also help understanding why `getNext` is just wrapping another function.

Reply:
Yeah, I don't see the point in the `getNext` function if all its doing it calling `getNextPath`.

Reply:
I've renamed `getNextPath` to `getNextNavigablePath`. There's good reason for `getNext` to continue its existence though. Otherwise this sort of configuration would be quite a bit more complex:
```
    getNextPath: getNext,
```

Reply:
@YetAnotherMatt you could always inline that function:
```
getNextPath : (req, steps) => getNextNavigablePath(req.path, req, steps)
```

Reply:
What's the verdict on whether I need to do so? (I pushed the latest changes before reading this final comment)

Reply:
I think inlining is a good idea - it will still be a single line and I find the function wrapping an unnecessary abstraction.